src/ClassDiscovery.php

```diff
@@ -163,18 +163,18 @@ protected static function evaluateCondition($condition)
             // Should be extended for functions, extensions???
             return class_exists($condition);
         } elseif (is_callable($condition)) {
-            return $condition();
+            return (bool) $condition();
         } elseif (is_bool($condition)) {
             return $condition;
         } elseif (is_array($condition)) {
             $evaluatedCondition = true;
 
             // Immediately stop execution if the condition is false
             for ($i = 0; $i < count($condition) && false !== $evaluatedCondition; ++$i) {
-                $evaluatedCondition &= static::evaluateCondition($condition[$i]);
+                $evaluatedCondition = $evaluatedCondition && static::evaluateCondition($condition[$i]);
```

Comment:
> the loop already has a check to abort immediately

No, since `evaluatedCondition` might be integer 0. And we do strict comparison for false `false !== $evaluatedCondition`. 

Sure. I can simplify it more.